lib/nats.js

```diff
@@ -831,10 +843,7 @@ Client.prototype.initState = function() {
  * @api public
  */
 Client.prototype.close = function() {
-    if (this.pingTimer) {
-        clearTimeout(this.pingTimer);
-        delete this.pingTimer;
-    }
+    this.cleanupTimers();
```

Comment:
this is not needed

Reply:
why?

Reply:
close on the stream will do it.

Reply:
Ok, we will still have all the subscriptions by that time, right? That is, there is no place where we would remove them by the time we get to stream.close and therefore are unable to cancel subscriptions' timer?

Reply:
actually - my bad the cleanup is required as closing the stream from `close` will not emit the 'close' event on the stream.